src/FileGetContents.php

```diff
@@ -69,33 +111,56 @@ public static function getNextRequestHeaders()
         return $return;
     }
 
+    /**
+     * @param resource|array|null $context A valid context resource created with `stream_context_create()`. If you don't
+     *                                     need to use a custom context, you can skip this parameter.
+     *
+     * @return resource|array|null Context to which the headers has been set.
+     *
+     * @private since 1.1.0
+     *
+     * TODO (2.0.0): change the name to reflect on the immutable side.
+     */
     public static function setHttpHeaders($context)
     {
         $headers = self::getNextRequestHeaders();
-        if (!empty($headers)) {
-            $options = stream_context_get_options($context);
-            if (!isset($options['http'])) {
-                $options['http'] = array('header'=>array());
-            } elseif (!isset($options['http']['header'])) {
-                $options['http']['header'] = array();
-            } elseif (is_string($options['http']['header'])) {
-                $options['http']['header'] = explode("\r\n", $options['http']['header']);
-            }
-            $headers = empty($options['http']['headers']) ? $headers : array_merge($options['http']['headers'], $headers);
-            stream_context_set_option(
-                $context,
-                'http',
-                'header',
-                $headers
-            );
+
+        if (empty($headers)) {
```

Comment:
the changes are due to that line, i.e. doing an early return to avoid a 2 deep nested statements